### ISSUE TITLE
Refactor out usage of `page.html`

### ISSUE THREAD

**Issue by angela-tran** [cal-itp/benefits#775]
Refactor templates that extend `page.html` so that they instead define their own structure. Templates can still make use of our `viewmodel` classes if that is helpful. Then, delete `page.html`.

If it makes sense to open a separate PR for each template that needs refactoring, feel free to do so.

## Acceptance Criteria

- [ ] No user-facing changes -- all pages should look and function as before
- [ ] `page.html` no longer exists

## Additional context
 - `page.html` was intended to be a generic and reusable Django template that all our pages could extend. However, recently we've noticed it gets in the way more often than it helps. For example, in [#403](https://github.com/cal-itp/benefits/pull/403/files#diff-70ed6fd5fa2321e83db0d63df8312f680cd40428de4ada50dab22e93f55db7c7), we made `help.html` override a lot of `page.html` in order to achieve a more customized structure that was needed by Login.gov integration.
 - In tickets like [#374](https://github.com/cal-itp/benefits/issues/374) and [#516](https://github.com/cal-itp/benefits/issues/516), we started a pattern of having specific template files for specific pages. This is easier to read and modify.
 - We can still reuse code via [include](https://docs.djangoproject.com/en/3.2/ref/templates/builtins/#include)s.
 - In #668, you can see our shift towards this approach when https://github.com/cal-itp/benefits/commit/dd4b0198cbd6dc186afbfba93df9b29967172c52 removed `PageTemplateResponse`.

Here are all the templates that currently extend `page.html`:
```
benefits/core/templates/core/agency_index.html
benefits/core/templates/core/help.html
benefits/eligibility/templates/eligibility/confirm.html
benefits/eligibility/templates/eligibility/start.html
benefits/eligibility/templates/eligibility/unverified.html
benefits/enrollment/templates/enrollment/index.html
benefits/enrollment/templates/enrollment/retry.html
benefits/enrollment/templates/enrollment/success.html
benefits/templates/400.html
benefits/templates/404.html
benefits/templates/500.html
```

**Comment by thekaveman** [comment]
I think this can either be closed and/or refactored into #961?

**Comment by angela-tran** [comment]
@thekaveman Yep, I will go ahead and close this one.
